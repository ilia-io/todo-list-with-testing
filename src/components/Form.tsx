import React, { useState } from 'react';

interface IFormProps {
  addTodo: (addInput: string) => void
}

const Form: React.FC<IFormProps> = ({ addTodo }) => {
  const [addInput, setAddInput] = useState('');

  const onClickAdd = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    addTodo(addInput);
    setAddInput('');
  };

  return (
    <form
      data-testid="form"
      onSubmit={(event) => {
        onClickAdd(event);
      }}
    >
      <input
        onChange={(event) => setAddInput(event.target.value)}
        value={addInput}
        className="input-todo"
        type="text"
        placeholder="Что будем делать?"
      />
      <button data-testid='add-btn' type="submit">Добавить +</button>
    </form>
  );
};

export default Form;
