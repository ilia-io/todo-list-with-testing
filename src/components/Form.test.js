import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Form from './Form';

describe('Form component tests', () => {
  test('renders form', () => {
    render(<Form />);
    const form = screen.getByTestId('form');
    expect(form).toBeInTheDocument();
    expect(form).toMatchSnapshot();
  });
  test('renders input', () => {
    render(<Form />);
    const input = screen.getByPlaceholderText(/Что будем делать?/i);
    expect(input).toBeInTheDocument();
  });
  test('button function', () => {
    let testText = '';
    render(
      <Form
        addTodo={(text) => {
          testText = text;
        }}
      />
    );
    const input = screen.getByPlaceholderText(/Что будем делать?/i);
    const btn = screen.getByRole('button');
    userEvent.type(input, 'write tests');
    fireEvent.click(btn);
    expect(testText).toBe('write tests');
  });
});
