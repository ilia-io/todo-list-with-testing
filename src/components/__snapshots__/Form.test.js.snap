// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component tests renders form 1`] = `
<form
  data-testid="form"
>
  <input
    class="input-todo"
    placeholder="Что будем делать?"
    type="text"
    value=""
  />
  <button
    data-testid="add-btn"
    type="submit"
  >
    Добавить +
  </button>
</form>
`;
